oe_editorial.module

```diff
@@ -6,3 +6,52 @@
  */
 
 declare(strict_types = 1);
+
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Url;
+
+/**
+ * Implements hook_field_widget_form_alter().
+ */
+function oe_editorial_field_widget_form_alter(&$element, FormStateInterface $form_state, $context) {
+  /** @var \Drupal\Core\Field\FieldDefinitionInterface $field_definition */
+  $field_definition = $context['items']->getFieldDefinition();
+
+  // We add an after build method so we can alter the Url of the help link.
+  if (in_array($field_definition->getType(), [
+    'text',
+    'text_long',
+    'text_with_summary',
+  ])) {
+    $element['#after_build'][] = '_oe_editorial_alter_textarea_help';
+  }
+}
+
+/**
+ * After build callback to alter the Url of the help link on text areas.
+ */
+function _oe_editorial_alter_textarea_help($form_element, FormStateInterface $form_state) {
+  if (!isset($form_element['format']['format'])) {
+    return $form_element;
+  }
+  foreach ($form_element['format']['format']['#options'] as $allowed_format_id => $allowed_format_name) {
+    // We add one link for each of the available format types.
+    $form_element['format']['help'][$allowed_format_id] = [
+      '#type' => 'link',
+      '#title' => t('About the :format_name format', [':format_name' => $allowed_format_name]),
+      '#url' => Url::fromRoute('filter.tips', ['filter_format' => $allowed_format_id]),
+      '#attributes' => [
+        'class' => [
+          'filter-help-item',
+          'filter-help-' . $allowed_format_id,
+        ],
+        'target' => '_blank',
+      ],
+    ];
+  }
+  // Remove the default static link.
+  unset($form_element['format']['help']['about']);
+  $form_element['format']['help']['#attached']['library'][] = 'oe_editorial/help';
```

Comment:
If possible, I think that it could be better to use altering of original library.